rust/exports.c

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// A hack to export Rust symbols for loadable modules without having to redo
+// the entire `include/linux/export.h` logic in Rust.
+//
+// This requires the Rust's new/future `v0` mangling scheme because the default
+// one ("legacy") 1) uses a hash suffix which cannot be predicted across
+// compiler versions and 2) uses invalid characters for C identifiers
```

Comment:
I missed this, but the `v0` mangling scheme is also unpredictable across compiler versions. The symbol hash at the end has been replaced with a crate id near the start of the symbol. For example in `_RNvCs21hi0yVfW1J_8rust_out4main` the crate id is `Cs21hi0yVfW1J`.

Reply:
Thanks, I will remove the first part of the sentence. It is actually a bit misleading, because the "hard" requirement is the second one: mixing `rustc` compiler versions is not something we will support. IIRC I was thinking about possible patch level fixes to `rustc` stable versions when I wrote this.